cdc/scheduler/internal/tp/table.go

```diff
@@ -0,0 +1,355 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tp
+
+import (
+	"context"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	"github.com/pingcap/tiflow/cdc/model"
+	"github.com/pingcap/tiflow/cdc/processor/pipeline"
+	"github.com/pingcap/tiflow/cdc/scheduler/internal"
+	"github.com/pingcap/tiflow/cdc/scheduler/internal/tp/schedulepb"
+	"go.uber.org/zap"
+)
+
+// table is a state machine that manage the table's state,
+// also tracking its progress by utilize the `TableExecutor`
+type table struct {
+	id model.TableID
+
+	status   schedulepb.TableStatus
```

Comment:
Why not just `schedulepb.TableState`?

Reply:
since this is also used to build `heartbeatResponse`, which needs checkpoint.

`status` is more useful in such use case.

Reply:
What about query `TableMeta()` when building `HeartbeatResponse`? In this way, we don't need `refresh` to reconcile checkpoint and resolved ts.

Reply:
but we still need reconcile state. I don't think there is any difference.